libraries/AP_LocationDB/tests/test_database.cpp

```diff
@@ -0,0 +1,249 @@
+#include <AP_gtest.h>
+
+/*
+  tests for AP_LocationDB
+ */
+
+#include <AP_LocationDB/AP_LocationDB.h>
+#include <GCS_MAVLink/GCS.h>
+
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL || CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+
+static float random_float(void) {
+    return ((((unsigned)random()) % 2000000) - 1.0e6) / 1.0e6;
+}
```

Comment:
We have a HAL function to get random numbers, I believe.

But... we shouldn't be using random numbers in tests like this unless it is strictly required.  Repeatability is key.  *other* sorts of tests can monte-carlo all they like, but these unit tests shouldn't change run-to-run.

Which is to say you should be testing codepaths with crafted identifiers here, not relying on random numbers to explore the space.

Reply:
Okay. I have removed the use of random numbers from the tests. Thanks for the suggestion.